src/hooks/safe-apps/useRankedSafeApps.ts

```diff
@@ -17,13 +19,22 @@ const useRankedSafeApps = (safeApps: SafeAppData[], pinnedSafeApps: SafeAppData[
       .filter(Boolean) as SafeAppData[]
 
     // we add random Safe Apps if no enough Safe Apps are present
-    const numberOfRandomSafeApps = NUMBER_OF_SAFE_APPS - mostUsedSafeApps.length
     const nonRankedApps = safeApps.filter((app) => !mostUsedSafeAppsIds.includes(String(app.id)))
-    const randomSafeApps = sampleSize(nonRankedApps, numberOfRandomSafeApps)
+    const randomSafeApps = sampleSize(nonRankedApps, NUMBER_OF_SAFE_APPS)
 
-    const rankedSafeApps = [...mostUsedSafeApps, ...randomSafeApps]
+    const rankedSafeApps = mostUsedSafeApps.concat(randomSafeApps)
 
-    return rankedSafeApps
+    // Display the safe-claiming-app at the first position
```

Comment:
Can we do it only for Safes that actually have something to claim/already claimed?
Might be not very interesting or even annoying to show it to not eligible users.

Reply:
I think we should leave it as is to match the widget. We'd otherwise have to fetch the total allocation. We can always remove the tag if/when needed to remove it from the top position.